### PR TITLE
[macOS] Merge FlutterMetalCompositor, superclass

### PR DESCRIPTION
**Pull request by cbracken**
Now that OpenGL support in Flutter on macOS has been removed, the only FlutterCompositor implementation is the Metal-based compositor. This patch merges it into the FlutterCompositor base class since there's no longer any reason to keep the interface separate from its implementation.

This patch will be followed by similar patches for the renderer, surface manager, and external textures.

Issue: https://github.com/flutter/flutter/issues/108304
Issue: https://github.com/flutter/flutter/issues/114445

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/macos/framework/Source/FlutterCompositor.h [hunk 1]
```diff
@@ -25,52 +26,57 @@ class FlutterCompositor {
   // The view_provider is used to query FlutterViews from view IDs,
   // which are used for presenting and creating backing stores.
   // It must not be null, and is typically FlutterViewEngineProvider.
-  explicit FlutterCompositor(id<FlutterViewProvider> view_provider);
+  explicit FlutterCompositor(
+      id<FlutterViewProvider> view_provider,
+      FlutterPlatformViewController* platform_views_controller,
+      id<MTLDevice> mtl_device);
 
   virtual ~FlutterCompositor() = default;
```
Comment (knopp) [inline shell/platform/darwin/macos/framework/Source/FlutterCompositor.h:34]: This doesn't need to be virtual.

File: shell/platform/darwin/macos/framework/Source/FlutterCompositor.h [hunk 2]
```diff
@@ -25,52 +26,57 @@ class FlutterCompositor {
...
-      const FlutterBackingStore* backing_store) = 0;
+  // Releases the memory for any resources that were allocated for the
+  // specified backing store.
+  virtual bool CollectBackingStore(const FlutterBackingStore* backing_store);
 
   // Presents the FlutterLayers by updating the FlutterView specified by
-  // `view_id` using the layer content.
-  // Present sets frame_started_ to false.
+  // `view_id` using the layer content. Sets frame_started_ to false.
   virtual bool Present(uint64_t view_id,
```
Comment (knopp) [inline shell/platform/darwin/macos/framework/Source/FlutterCompositor.h:57]: Doesn't need to be virtual any more.

File: shell/platform/darwin/macos/framework/Source/FlutterCompositor.h [hunk 3]
```diff
@@ -25,52 +26,57 @@ class FlutterCompositor {
...
+  // If the backing store is being requested for the first time for a given
+  // frame, this compositor does not create a new backing store but rather
+  // returns the backing store associated with the FlutterView's
+  // FlutterSurfaceManager.
+  //
+  // Any additional state allocated for the backing store and saved as
+  // user_data in the backing store must be collected in the backing_store's
+  // destruction_callback field which will be called when the embedder collects
+  // the backing store.
   virtual bool CreateBackingStore(const FlutterBackingStoreConfig* config,
```
Comment (knopp) [inline shell/platform/darwin/macos/framework/Source/FlutterCompositor.h:48]: Doesn't need to be virtual.

**Review by knopp — APPROVED** [review]
Just some nits.

**Comment by cbracken** [comment]
Nice catch!

**Pull request merged**